### PR TITLE
Add plot for runtime benchmarking

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

## Motivation

This PR is to add a plot to compare multiple samplers' runtime.

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@y0z 

Probably, it is nicer to have a guidance to pre-commit in either PR template.
For example, something like this is sufficient.

To merge this PR, please use `pre-commit`. 

```shell
$ <Install command>
$ pre-commit run --all-files
```

File: package/visualization/plot_sampling_speed/__init__.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import numpy as np
+import optuna
+from scipy.stats import linregress
+
+
+if TYPE_CHECKING:
+    from matplotlib.axes._axes import Axes
+
+
+def objective(trial: optuna.Trial) -> float:
```
Comment (y0z) [inline package/visualization/plot_sampling_speed/__init__.py:16]: Perhaps this objective function can be removed.

**Review by y0z — COMMENTED** [review]
I leave a minor comment.

File: package/visualization/plot_sampling_speed/__init__.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    x = trial.suggest_float("x", -5, 5)
+    return x**2
+
+
```
Comment (nabenabe0928) [inline package/visualization/plot_sampling_speed/__init__.py:20]:
```suggestion
```

**Review by y0z — APPROVED** [review]
LGTM

**Pull request merged**